### PR TITLE
Move d34raphael into plugin.

### PR DESCRIPTION
**Pull request by mhemesath**
I went ahead and moved the d34raphael code into its own project.  D3 is now referenced as submodule. This enables d34raphael to quickly pull in updates form d3 and keep the custom plugin code separate at the same time.

### CODE REVIEW

**Comment by webmonarch** [comment]
Hey Mike, thanks!  I will take a look tonight and review.  

I agree in general with the architecture.  The only thing I was concerned about (though I didn't look specifically) is that I delegate to d3 code as much as possible to implement the selector methods that don't need to be customized for Raphael.  

As long as we can keep that and keep the d3 testing infrastructure, I am happy.

**Comment by mhemesath** [comment]
Yeah, we can definitely iterate on it.  The tests are currently all contained within the D3 submodule, but we could probably setup a make command to auto-run the D3 Test suite with the d34raphael version of the code.

**Comment by mhemesath** [comment]
I was able to get selection via classname working with the help of sizzle. I also added remove() to the selection prototype to be used with exit().

There are quite a bit more features I plan on adding in the near future (e.g. transition support).  Until we get this merged to the original repo, I'll keep pushing them into this fork.

**Comment by webmonarch** [comment]
Hey Mike,

I just got a chance to review your changes.  Right now, I am just focusing on the first commit, the restructuring one.  

I need to mull a bit more before figuring out how to move forward on master, but am I correct in understanding the d34raphael.v2.js is completely self-contained with both d3 and raphael included internally.  So, on an client HTML page, we would only need to include that one JS file?

This is less architecturally different than I had in mind...and more of a reorg removing the separate (and admittedly confusing) two repository situation...

Did I understand the intention correctly?

**Comment by mhemesath** [comment]
Yeah, there is still only one JS file. A custom build of D3 is unavoidable as we need to disable a few native API's (selectors, transitions) for it to work.

File: src/raphael/customattributes.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    function d3_raphael_addCustomAttributes(paper) {
+        paper.ca.d = function(path) {
+            return {path: path};
+        };
+    }
+
+    // adds selection.attr("class",...) support
+    // see raphaelselction.attr
+
+    Raphael.el.setAttribute = function(name, value) {
```
Comment (mhemesath) [inline src/raphael/customattributes.js:16]: Enabling D3 transitions was simply a matter of exposing the Element.attribute methods on Raphael elements.

**Comment by mhemesath** [comment]
Hey Eric, are you interested in accepting this pull request or do you have reservations?  If you do have reservations, I'd like to discuss them so we move forward with this project. Thanks!

**Comment by webmonarch** [comment]
Hey Mike,

I really like the functionality you are enabling!  My main concrete objection right now is we are losing the vows tests for some of the new functionality.  Also, if we can achieve the same cleanliness of building from a stock raphael like you do now from a stock d3...I think I would be able to pull the lever on this.

Let me know if that doesn't make sense!

**Comment by mhemesath** [comment]
Boom! Updated to include passing tests on TravisCI :p

**Comment by mhemesath** [comment]
In regards to Raphael, you want me to include it in the project as a submodule and concatenate it to d34raphael or do you prefer we leave it as an external file?

**Comment by issa-tseng** [comment]
@mhemesath: given you have r2d3 going on now, and this project is mostly in maintenance mode, I'm assuming you won't mind if I close out this thread?

**Pull request closed without merging**